### PR TITLE
Card: refactor subcomponents folder structure

### PR DESCRIPTION
**Pull request by ciampo**
## Description

_This is the first of a series of PRs with the final goal of achieving the same result as #32243 in smaller (and easier to review) steps._

In this PR the folder structure of `<Card />` and its subcomponents is changed so that each subcomponent has a dedicated folder. All of the subcomponents are then exported in the `index.js` file. There shouldn't be any changes to the actual implementation.

This PR will enable the next PRs to focus on swapping the current implementation with the new one (_g2_), without the noise caused by the folder structure changes.

## Testing instructions

There are no changes to runtime code.

- [ ] Project builds correctly
- [ ] All card unit tests pass
- [ ] Run the project locally and compare the `Components / Card` stories against the [production version](https://wordpress.github.io/gutenberg/?path=/story/components-card--default) — there shouldn't be any difference

## Screenshots 

N/A

## Types of changes

Refactor (folder structure)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- N/A I've tested my changes with keyboard and screen readers. 
- N/A My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- N/A I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/card/test/index.js [hunk 1]
```diff
@@ -6,12 +6,7 @@ import { mount, shallow } from 'enzyme';
 /**
  * Internal dependencies
  */
-import Card from '../';
-import CardBody from '../body';
-import CardDivider from '../divider';
-import CardFooter from '../footer';
-import CardHeader from '../header';
-import CardMedia from '../media';
+import { Card, CardBody, CardDivider, CardFooter, CardHeader, CardMedia } from '../';
```
Comment (tyxla) [inline packages/components/src/card/test/index.js]: Seems like those need to go on separate lines - the linter complains 😉

File: packages/components/src/index.js [hunk 2]
```diff
@@ -25,12 +25,7 @@ export { default as BaseControl } from './base-control';
 export { default as __experimentalBoxControl } from './box-control';
 export { default as Button } from './button';
 export { default as ButtonGroup } from './button-group';
-export { default as Card } from './card';
-export { default as CardBody } from './card/body';
-export { default as CardDivider } from './card/divider';
-export { default as CardFooter } from './card/footer';
-export { default as CardHeader } from './card/header';
-export { default as CardMedia } from './card/media';
+export { Card, CardBody, CardDivider, CardFooter, CardHeader, CardMedia } from './card';
```
Comment (tyxla) [inline packages/components/src/index.js]: Seems like those need to go on separate lines - the linter complains 😉

File: packages/components/src/card/card/index.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (tyxla) [inline packages/components/src/card/card/index.js:1]: Do we really want to put this in its own directory? `/card/card` seems a bit odd 😉  Should we move this component a directory up?

**Review by tyxla — CHANGES_REQUESTED** [review]

File: packages/components/src/card/card/index.js (hunk 3 above)
Comment (ciampo) [reply packages/components/src/card/card/index.js:1]: This folder structure was agreed with @sarayourfriend. The same approach was used for the `<Flex />` component which sits in the [`/packages/components/src/flex/flex` folder](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/flex/flex)

Comment (tyxla) [reply packages/components/src/card/card/index.js:1]: Consistency FTW 👍  Let's go with that then!

File: packages/components/src/card/test/index.js (hunk 1 above)
Comment (ciampo) [reply packages/components/src/card/test/index.js]: Good catch! Somehow the code was not automatically formatted.

Fixed in https://github.com/WordPress/gutenberg/pull/32557/commits/564c683c2a3ff5321d39265075555b995f0746e5

File: packages/components/src/index.js (hunk 2 above)
Comment (ciampo) [reply packages/components/src/index.js]: Fixed in https://github.com/WordPress/gutenberg/pull/32557/commits/564c683c2a3ff5321d39265075555b995f0746e5

**Review by tyxla — APPROVED** [review]
LGTM 👍

**Pull request merged**